e2e/scenario_helpers_test.go

```diff
@@ -107,6 +111,9 @@ func createAndValidateVM(ctx context.Context, t *testing.T, scenario *Scenario)
 	vmssName := getVmssName(t)
 	createVMSS(ctx, t, vmssName, scenario, privateKeyBytes, publicKeyBytes)
 
+	err = getCustomScriptExtensionStatus(ctx, t, *scenario.Runtime.Cluster.Model.Properties.NodeResourceGroup, vmssName)
+	require.NoError(t, err)
```

Comment:
Maybe other tests would benefit from it as well?
What if it's moved into `creteVMSS` function?

Reply:
It will also validate the behaviour hasn't changed.

Reply:
yes currently this will get run for all tests, there is no check for scriptless nodes. I think it makes more sense to keep it here since it is a validation step.

Reply:
ah, sorry. I thought I'm looking at different file.